python/cugraph-pyg/cugraph_pyg/utilities/api_tools.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import warnings
+import inspect
+import types
+
+experimental_prefix = "EXPERIMENTAL"
+
+
+def experimental_warning_wrapper(obj):
+    """
+    Wrap obj in a function or class that prints a warning about it being
+    "experimental" (ie. it is in the public API but subject to change or
+    removal), prior to calling obj and returning its value.
+
```

Comment:
Curious on why this is not imported from cugraph ?

Reply:
`cugraph` is not a hard dependency of `cugraph-pyg`.  It may not be installed.